### PR TITLE
rework convert_attribute docstring

### PR DESCRIPTION
**Pull request by jw3126**
supersedes https://github.com/MakieOrg/Makie.jl/pull/2504

### CODE REVIEW

File: MakieCore/src/conversion.jl [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+
+# Example
+```jldoctest
+julia> using Makie
+
+julia> Makie.convert_attribute(:black, key"color"())
+RGBA{Float32}(0.0f0,0.0f0,0.0f0,1.0f0)
+```
+"""
 function convert_attribute end
```
Comment (jw3126) [inline MakieCore/src/conversion.jl:17]: I have a couple of questions about this function. Is this an internal function? Is the user supposed to overload it? If so are there typical cases, when a user might want to overload this?

Comment (jw3126) [reply MakieCore/src/conversion.jl:17]: For instance I might have a custom recipe `particles` that takes attributes
`photoncolor` and `electroncolor.
Should I overload 
```julia
MakieCore.convert_attributes(value, ::key"photoncolor", ::key"phsp") = convert_attributes(value, ::key"color")
MakieCore.convert_attributes(value, ::key"electroncolor", ::key"phsp") = convert_attributes(value, ::key"color")
```
?

Comment (SimonDanisch) [reply MakieCore/src/conversion.jl:17]: No, this is only for atomic plot types (like scatter, mesh, etc) and gets called in the backend.
You can use `to_color` in your recipe for such cases...
Overloading is only useful, if you have your own custom color type and want to use it directly as a color attribute!

**Comment by SimonDanisch** [comment]
Thanks!

**Pull request merged**